doc/nrf/ug_bootloader_adding.rst

```diff
@@ -0,0 +1,350 @@
+.. _ug_bootloader_adding:
+
+Adding a bootloader chain
+#########################
+
+.. contents::
+   :local:
+   :depth: 2
+
+You can add bootloader chains to an application in the following ways:
+
+* Permanently:
+
+  * Using Kconfig fragments.
+  * Using ``prj.conf`` files.
+
+* Temporarily, for a single build command, using :ref:`one-time CMake arguments <zephyr:west-building-cmake-args>`.
+
+Kconfig fragments and ``prj.conf`` files use the same syntax and they are the recommended method.
+You can add bootloader chains to nearly any sample in |NCS| or Zephyr for rapid testing and experimentation of their capabilities.
+
+The type of bootloader that you should add is always dependent upon application needs.
+
+.. _ug_bootloader_adding_immutable:
+
+Adding an immutable bootloader
+******************************
+
+The following sections describe how to add either |NSIB| or MCUboot as immutable bootloaders.
+
+.. _ug_bootloader_adding_immutable_b0:
+
+Adding |NSIB| as an immutable bootloader
+========================================
+
+To build |NSIB| with a Zephyr or |NCS| sample, enable the :option:`CONFIG_SECURE_BOOT` in the application's ``prj.conf`` file, in an associated Kconfig fragment, or using the command line:
+
+.. code-block:: console
+
+   west build -b nrf52840dk_nrf52840 zephyr/samples/hello_world -- -DCONFIG_SECURE_BOOT=y
+
+|how_to_configure|
+
+Like other child images, :ref:`image-specific configurations <ug_multi_image_variables>` can be assigned at build time to further customize the bootloader's functionality.
+
+To ensure that the immutable bootloader occupies as little flash memory as possible, you can also apply the :file:`prj_minimal.conf` configuration:
+
+.. code-block:: console
+
+   west build -b nrf52840dk_nrf5840 zephyr/samples/hello_world -- \
+   -DCONFIG_SECURE_BOOT=y \
+   -Db0_CONF_FILE=prj_minimal.conf
+
+See :ref:`ug_bootloader_config` for more information about using Kconfig fragments.
+
+.. _ug_bootloader_adding_immutable_keys:
+
+Adding a custom signature key file
+----------------------------------
+
+To add a signature key file to this bootloader, set the :option:`CONFIG_SB_SIGNING_KEY_FILE` option in the application's ``prj.conf`` file, in an associated Kconfig fragment, or using the command line:
+
+.. tabs::
+
+   .. group-tab:: Kconfig / prj.conf
+
+      .. code-block:: console
+
+         CONFIG_SB_SIGNING_KEY_FILE="priv.pem"
+
+   .. group-tab:: Command line
+
+      .. code-block:: console
+
+         -DCONFIG_SB_SIGNING_KEY_FILE=\"priv.pem\"
+
+      Escaped quotations avoid malformed-string warnings from Kconfig.
+
+This option only accepts the private key of an ECDSA key pair, as the public key will be extracted from it automatically during build time.
```

Comment:
```suggestion
This option only accepts the private key of an ECDSA key pair, as the public key is extracted from it automatically at build time.
```

What extracts the key?

Reply:
Reworded, please see the latest changes.